### ISSUE TITLE
NIRSpec TSO filenames may be missing the slitname

### ISSUE THREAD

**Issue by stscijgbot-jp** [spacetelescope/jwst#8688]
_Issue [JP-3702](https://jira.stsci.edu/browse/JP-3702) was created on JIRA by [Tyler Pauly](https://jira.stsci.edu/secure/ViewProfile.jspa?name=tpauly):_

DMS Ops reported a data product has had its filename change:

<https://stsci.slack.com/archives/C9KF4MVEV/p1722348421168879>
```
I just noticed an issue where a JWST observation that had been included in a DOI was no longer available in the MAST portal. The original observation was:
jw02512-o008_t007_nirspec_f290lp-g395h-s1600a1-sub2048
and the observation that I think is equivalent is:
jw02512-o008_t007_nirspec_f290lp-g395h-sub2048
```

It looks like work completed for JP-3233 may have altered behavior for NIRSpec TSO data - it was not given the new product name generation, and the change to _get_opt_element removed the slit name from the concatenated list of elements. 

Work should include new test coverage of NIRSpec BOTS - this should have been caught by regressions.

**Comment by stscijgbot-jp** [comment]
_Comment by [Melanie Clarke](https://jira.stsci.edu/secure/ViewProfile.jspa?name=mclarke) on [JIRA](https://jira.stsci.edu/browse/JP-3702?focusedCommentId=715447#comment-715447):_

Fixed by #8699

**Comment by stscijgbot-jp** [comment]
_Comment by [Tyler Pauly](https://jira.stsci.edu/secure/ViewProfile.jspa?name=tpauly) on [JIRA](https://jira.stsci.edu/browse/JP-3702?focusedCommentId=939037#comment-939037):_

Tested with 1.17.1 and confirmed that the slitname is present in the output for jw06456-o001, which shows the regressed behavior in MAST archived products:

 !Screenshot 2025-01-09 at 9.19.03 AM.png|thumbnail!
